### ISSUE TITLE
Squirrel Startup Exe Icon missing

### ISSUE THREAD

**Issue by AzonInc** [Squirrel/Squirrel.Windows#1790]
I'm generating Setup Files witrh the Electron-Forge Squirrel Maker.
However after trying several things and seraching for similar Issues, its not possible to get the App Icon working.

It's still the default Package Icon, which shouldn't be there (see: Restaurant Companion.exe)
![image](https://user-images.githubusercontent.com/11911587/153718841-8a132225-df74-459a-8412-dc539d16e378.png)
The Packaged App has the correct Icon assigned (Restaurant Companion.exe, squirrel.exe):
![image](https://user-images.githubusercontent.com/11911587/153718870-a08aa02e-2e6f-4e06-8639-7c7d92cea0b5.png)
The Control Panel Icon is also correct:
![image](https://user-images.githubusercontent.com/11911587/153719364-9a9452cd-775c-4959-bbf0-cd296a2f3ecf.png)

My Forge Config:
```
const path = require('path');
const package = require('./package.json');

module.exports = {
  packagerConfig: {
    asar: true,
    icon: path.resolve(__dirname, 'src', 'assets', 'images', 'icon.ico')
  },
  makers: [
    {
      name: '@electron-forge/maker-squirrel',
      platforms: ['win32'],
      config: (arch) => {
        return {
          name: 'RestaurantCompanion',
          authors: 'Brogli Informatik GmbH',
          exe: 'Restaurant Companion.exe',
          noMsi: true,
          remoteReleases: '',
          loadingGif: path.resolve(__dirname, 'src', 'assets', 'images', 'setup-animation.gif'),
          setupExe: `RestaurantCompanion-${package.version}-setup-${arch}.exe`,
          setupIcon: path.resolve(__dirname, 'src', 'assets', 'images', 'icon.ico'),
          iconUrl: "https://REMOVED/restaurant-companion/icon.ico",
          signWithParams: process.env['WINDOWS_CODESIGN_ARGS']
        }
      }
    },
    {
      name: '@electron-forge/maker-zip',
      platforms: ['darwin', 'win32']
    },
    {
      name: '@electron-forge/maker-deb',
      platforms: ['linux']
    },
    {
      name: '@electron-forge/maker-rpm',
      platforms: ['linux']
    }
  ]
};
```

**Comment by easternmotors** [comment]
@AzonInc have you figured out what was going on yourself? I've been looking into it and may have some insight. What does your `package.json` look like? Specifically the `productName` and `name` fields.

**Comment by AzonInc** [comment]
Hi, @easternmotors .
I was not able to get it working yet.

Here the relevant package json part:
`{
  "name": "restaurant-companion",
  "productName": "Restaurant Companion",
  "version": "1.6.0-beta.5",
  "description": "Restaurant CMS - Companion App",
  "main": "src/main.js",
  "scripts": {
    "install": "electron-rebuild",
    "start": "electron-forge start",
    "package": "electron-forge package",
    "make": "electron-forge make --platform win32",
    "publish": "electron-forge publish",
    "version": "conventional-changelog -p angular -i CHANGELOG.md -s && git add CHANGELOG.md"
  },
  "keywords": [],
  "author": {
    "name": "BIT GmbH",
    "email": "info@bit.ch"
  },
  "license": "MIT",
  "config": {
    "forge": "./forge.config.js"
  },`

**Comment by thorsent** [comment]
Did anyone manage to get to the bottom of this? Best we can tell, there may be a stale icon in some registry entry somewhere.

**Comment by easternmotors** [comment]
@thorsent no, still experiencing this issue. It only seems to be an issue when one of the variable which has a fallback of `packageJSON.name` is overwritten (like `productName` in the example from @AzonInc above -- if not set it defaults to `packageJSON.name`). If I had to guess, the issue is that somewhere it is pulling the variable from `packageJSON.name` for a path instead of the custom name being set, causing a failed lookup -- but this is just a guess. I haven't had time in a while to dive into the root cause.

**Comment by lukasz-mical** [comment]
@AzonInc, I had a similar issue with the icon and upon doing some research, I discovered that the ID (name in package.json) should not contain dots or spaces. I had a dot in the name, so I changed it to a dash. However, this did not resolve the issue. Then, I tried removing all non-alphabetical characters, and after making this change, the icon started to work properly.

**Comment by lggarrison** [comment]
I followed @lukasz-mical's advice and removed the spaces from my `packagerConfig.name` in my `forge.config.js`

This fixed the setup.exe icon on Windows for me.

**Comment by thorsent** [comment]
My support team informs me that some antivirus programs may prevent rcedit from modifying the icon (perceiving it as an attack on an executable). Setting environment variable `DEBUG=*` causes rcedit to output logs which may help trace when this happens.

Not sure if this is therefore independent of the reported issues with spaces and other non-alphabetical characters.

**Comment by panda6412** [comment]
I've encountered an issue with my Electron app where it appears that a variable, which has a fallback to packageJSON.name, is not being correctly overwritten by productName. This is causing problems with path lookups and potentially stale icon entries in the Windows registry.

### Context
Here is the relevant part of my package.json:

```json
{
  "name": "deepfakeinspector",
  "productName": "Deepfake Inspector",
  "version": "1.0.2-dev",
  "description": "Stay ahead of deepfakes",
  "main": ".vite/build/index.js",
  "scripts": {
    "start": "electron-forge start",
    "package": "electron-forge package",
    "make": "electron-forge make",
    "make:win-arm64": "electron-forge make --arch=arm64",
    "publish": "electron-forge publish",
    "lint": "eslint --ext .ts,.tsx ."
  }
}
```
And here is my forge.config.ts:

```typescript
import type { ForgeConfig } from '@electron-forge/shared-types';
import { MakerSquirrel } from '@electron-forge/maker-squirrel';
import { MakerZIP } from '@electron-forge/maker-zip';
import { MakerDeb } from '@electron-forge/maker-deb';
import { MakerRpm } from '@electron-forge/maker-rpm';
import { VitePlugin } from '@electron-forge/plugin-vite';
import { FusesPlugin } from '@electron-forge/plugin-fuses';
import { FuseV1Options, FuseVersion } from '@electron/fuses';

import { APP_PROTOCOL_NAME } from './src/common/constants';

const config: ForgeConfig = {
  packagerConfig: {
    icon: 'public/images/icon',
    asar: false,
    protocols: [
      {
        name: 'Deepfake Inspector',
        schemes: [APP_PROTOCOL_NAME],
      },
    ],
  },
  rebuildConfig: {},
  makers: [
    new MakerSquirrel({
      setupIcon: 'public/images/icon.ico',
      iconUrl: 'https://d30nwnnt92yl6b.cloudfront.net/icon.ico',
      loadingGif: 'public/images/loading.gif',
      skipUpdateIcon: true,
    }),
    new MakerZIP({}, ['darwin']),
    new MakerRpm({}),
    new MakerDeb({}),
  ],
  plugins: [
    new VitePlugin({
      build: [
        {
          entry: 'src/main/index.ts',
          config: 'vite.main.config.ts',
        },
        {
          entry: 'src/preload/index.ts',
          config: 'vite.preload.config.ts',
        },
      ],
      renderer: [
        {
          name: 'main_window',
          config: 'vite.renderer.config.ts',
        },
      ],
    }),
    new FusesPlugin({
      version: FuseVersion.V1,
      [FuseV1Options.RunAsNode]: false,
      [FuseV1Options.EnableCookieEncryption]: true,
      [FuseV1Options.EnableNodeOptionsEnvironmentVariable]: false,
      [FuseV1Options.EnableNodeCliInspectArguments]: false,
      [FuseV1Options.EnableEmbeddedAsarIntegrityValidation]: true,
      [FuseV1Options.OnlyLoadAppFromAsar]: false,
    }),
  ],
};

export default config;
```

### Issue
The problem seems to arise when the productName variable is set. Despite this, somewhere in the build process, it appears to fall back to using packageJSON.name, causing a failed path lookup. This has led to issues such as stale icons in the registry.

### Attempts to Resolve
I have tried the following solutions without success:

- Running ie4uinit.exe -show in the command line and restarting File Explorer.
- Restarting the computer.
### Request for Help
Has anyone else encountered this issue? If so, how did you resolve it? Any insights or solutions would be greatly appreciated.

Thank you!

**Comment by nikwen** [comment]
@panda6412: If it's still helpful for you, I've documented a working setup here: https://github.com/electron-forge/electron-forge-docs/pull/209
